### ISSUE TITLE
Adding Bashlex fixes to upstream

### ISSUE THREAD

**Issue by petr-muller** [rh-lab-q/bkrdoc#18]
Zuzana did some good fixes of Bashlex but for a future we would not be able to maintain bashlex. So we would like to have these upgrades accepted in upstream

**Comment by petr-muller** [comment]
Is there something new with upstream?

[--- Commented from Asana.com
#commenter Jiří Kulda
---[aa]

**Comment by petr-muller** [comment]
Hi Zuzana,
I may found bug in your comment fix for bashlex(It's really may). I have problems with my markup version of bkrdoc in branch with bashlex. I think this problem is due your fixes. 

My problem is:
I need to parse single commands in which have tag at the end of a command line. For example: rlFail "Mangles the configuration value"#@. But when I give it to bashlex the output is [CommandNode(parts=[WordNode(parts=[] pos=(0, 6) word='rlFail'), WordNode(parts=[] pos=(7, 42) word='Mangles the configuration value#@')] pos=(0, 42))]
. As you can see the comment is still there but i think bashlex should erase it cause it does with all comments but not with this single one. When i put there a single white space bashlex erase it and everything is pretty fine. Could you please look at it?
Jirka

[--- Commented from Asana.com
#commenter Jiří Kulda
---[aa]

**Comment by blurrymoi** [comment]
I am not sure this is wrong,
if I have: rlRun "echo Hi"#@ 0 "test"  and run it in bash using beakerlib, the result is "test :: actually running 'echo Hi#@' " and does print Hi#@.

**Comment by petr-muller** [comment]
I see, i assume there should be a white space between something and beginning of a comment. OK Sorry for bothering. It was my mistake :smile:, but thank you for a quick response.

[--- Commented from Asana.com
#commenter Jiří Kulda
---[aa]

**Comment by petr-muller** [comment]
Hi Zuzana,
nowadays I'm trying to parse case and i found something strange. When you have a one lined case condition something like this: 

*)
    rlDie "Unknown syslog file!"
;;

te ast is: 
PatternNode(actions=[WordNode(parts=[] pos=(393, 398) word='rlDie'), WordNode(parts=[] pos=(399, 421) word='Unknown syslog file!')] pattern=[WordNode(parts=[] pos=(1362, 1363) word='*')] pos=(2350, 2397))

But when you have more lines, for example I add two the result is: 
PatternNode(actions=[CommandNode(parts=[WordNode(parts=[] pos=(393, 398) word='rlDie'), WordNode(parts=[] pos=(399, 421) word='Unknown syslog file!')] pos=(393, 421)), OperatorNode(op='\n' pos=(421, 422)), CommandNode(parts=[WordNode(parts=[] pos=(438, 443) word='rlDie'), WordNode(parts=[] pos=(444, 466) word='Unknown syslog file!')] pos=(438, 466)), OperatorNode(op='\n' pos=(466, 467))] pattern=[WordNode(parts=[] pos=(1362, 1363) word='*')] pos=(2350, 2443))

As you can see when there is a one line the resul in actions are list of wordnodes but when you have multiple lines the result is a list of CommandNodes. Is this a normal behavior or it's just a mistake and in the first case should be on command node?
Thanks Jiri

[--- Commented from Asana.com
#commenter Jiří Kulda
---[aa]

**Comment by blurrymoi** [comment]
Yes, I was aware of this and should probably make it somewhat consistent...

**Comment by blurrymoi** [comment]
Does this make sense or did you have something else in mind? :
``
case "$VALUE" in
        [yY][eE][sS]|"_")
            line1 param1 param2
            ;;
        *)
            line1 param1
	    line2 param3 param4
            ;;
    esac

``


``
CompoundNode(list=[
  CaseNode(pos=(0, 172), parts=[
    ReservedwordNode(pos=(0, 4), word='case'),
    WordNode(pos=(5, 13), word='$VALUE', parts=[
      ParameterNode(pos=(6, 12), value='VALUE'),
    ]),
    ReservedwordNode(pos=(14, 16), word='in'),
    PatternNode(actions=
      CommandNode(pos=(55, 74), parts=[
        WordNode(pos=(55, 60), word='line1'),
        WordNode(pos=(61, 67), word='param1'),
        WordNode(pos=(68, 74), word='param2'),
      ]), pattern=[
      WordNode(pos=(25, 37), word='[yY][eE][sS]'),
      WordNode(pos=(38, 41), word='_'),
    ], pos=(25, 74)),
    PatternNode(actions=[
      CommandNode(pos=(113, 125), parts=[
        WordNode(pos=(113, 118), word='line1'),
        WordNode(pos=(119, 125), word='param1'),
      ]),
      OperatorNode(op='\n', pos=(125, 126)),
      CommandNode(pos=(129, 148), parts=[
        WordNode(pos=(129, 134), word='line2'),
        WordNode(pos=(135, 141), word='param3'),
        WordNode(pos=(142, 148), word='param4'),
      ]),
      OperatorNode(op='\n', pos=(148, 149)),
    ], pattern=[
      WordNode(pos=(98, 99), word='*'),
    ], pos=(98, 125)),
    ReservedwordNode(pos=(168, 172), word='esac'),
  ]),
], pos=(0, 172))
``

**Comment by petr-muller** [comment]
Thanks this looks great. Is this update in your devel bashlex branch?

[--- Commented from Asana.com
#commenter Jiří Kulda
---[aa]

**Comment by blurrymoi** [comment]
It is now.

**Comment by petr-muller** [comment]
How we are with fixes delivered to upstream?

[--- Commented from Asana.com
#commenter Jiří Kulda
---[aa]

**Comment by blurrymoi** [comment]
I made no further attempts.
